src/test/java/seedu/taassist/logic/commands/UnassignCommandTest.java

```diff
@@ -0,0 +1,183 @@
+
+        List<Index> indices = new ArrayList<>();
+        indices.add(Index.fromOneBased(1));
+
+        UnassignCommand unassignCommand = new UnassignCommand(indices, moduleClass);
+        String expectedMessage = String.format(UnassignCommand.MESSAGE_SUCCESS, "index", indices, moduleClass);
+        Student expectedStudent = new StudentBuilder(ALICE).withModuleClasses().build();
+
+        CommandResult commandResult = unassignCommand.execute(modelStub);
+        assertEquals(expectedMessage, commandResult.getFeedbackToUser());
+        assertEquals(expectedStudent, modelStub.student);
+    }
+
+    @Test
+    public void execute_unassignMultipleStudents_success() throws CommandException {
+        ModelStubWithMultipleStudents modelStub = new ModelStubWithMultipleStudents();
+        ModuleClass moduleClass = new ModuleClass("GEN1000");
+
+        List<Index> indices = new ArrayList<>();
+        indices.add(Index.fromOneBased(1));
+        indices.add(Index.fromOneBased(2));
+
+        UnassignCommand unassignCommand = new UnassignCommand(indices, moduleClass);
+        String expectedMessage = String.format(UnassignCommand.MESSAGE_SUCCESS, "indices", indices, moduleClass);
+        Student expectedStudent1 = new StudentBuilder(ALICE).withModuleClasses().build();
+        Student expectedStudent2 = new StudentBuilder(BOB).withModuleClasses().build();
+
+        CommandResult commandResult = unassignCommand.execute(modelStub);
+        assertEquals(expectedMessage, commandResult.getFeedbackToUser());
+        assertEquals(expectedStudent1, modelStub.students.get(0));
+        assertEquals(expectedStudent2, modelStub.students.get(1));
+    }
+
+    /**
+     * A Model stub that pretends to have no module classes.
+     */
+    private class ModelStubWithNoModuleClasses extends ModelStub {
+
+        @Override
+        public boolean hasModuleClass(ModuleClass moduleClass) {
+            return false;
+        }
+
+        @Override
+        public ObservableList<ModuleClass> getModuleClassList() {
+            return FXCollections.observableArrayList();
+        }
+    }
+
+    /**
+     * A Model stub with one filtered student with an assigned class.
+     * Always says it has a module.
+     */
+    private class ModelStubWithOneStudent extends ModelStub {
+
+        private Student student = new StudentBuilder(ALICE).withModuleClasses("GEN1000").build();
+
+        @Override
+        public ObservableList<Student> getFilteredStudentList() {
+            return FXCollections.observableArrayList(student);
+        }
+
+        @Override
+        public void setStudent(Student target, Student editedStudent) {
+            student = editedStudent;
+        }
+
+        @Override
+        public boolean hasModuleClass(ModuleClass moduleClass) {
+            return true;
+        }
+    }
+
+    @Test
+    public void equals() {
+        ModuleClass moduleClass = new ModuleClass("GEN1000");
```

Comment:
I notice that the entire test case is constructed with the module class being forced to be "GEN1000". You might want to abstract it out as a constant instead, to retain better abstraction of the test cases and ease future modification.

Reply:
PR #129 by @njxue adds typical module classes and module class builders. I will update this PR after that gets merged.